azerothcore/azerothcore-wotlk
# fix(DB): Evening/Nighttime Game Events Update

Ever wondered why Miss Danna has her school children out looking at buildings in the middle of the night?  Doesn't Adam and Billy's fathers wonder where they are?  In vanilla wow they despawned at night.  

This PR will add evening and night events into Azerothcore.  Here are the changes:
Pyrewood Village Event Start Time fix -> Was too early
Moonrage Sentry Faction fix ->  Not hostile to alliance at night
Ultham Ironhorn -> Move to corral at night (Pyrewood Village Event)
Ultham's Rams -> Move to corral at night (Pyrewood Village Event)
Frost and Black Ram not available in WOTLK changed to correct Ram
Rotten Ghouls spawn during (Pyrewood Village) night event
Fleshrippers replace Rotten Ghouls when night event not active
Defias Smugglers replace Rotten Ghouls when night not active
Eastvale Children movement and locations adjusted
Evening Game event added -> Stars at 6:00 server time
Eastvale Children despawn during Evening event
Following Stormwind Civilians also despawn during evening event
Adam and Billy
Justin, Brandon, and Roman
Miss Danna and school children
Donna and William
Janey, Suzanne, and Lisan
Orphans that are running around at Orphanage
Karlee Chaddis, Paige Chaddis, and Gil
New Leprithus Event -> Starts at 8:00 server time
2 Leprithus spawn in different loactions in WOTLK at 8:00
Southern Graveyard has waypoints added
2 Rotten Ghoul minions spawn with each Leprithus

Some info for this update was pulled from TC database.



## Changes Proposed:
-  See above
-  

## Issues Addressed:

- Closes #7587

## SOURCE:

Here are some sources used:
https://wowpedia.fandom.com/wiki/Nocturnal
https://tbc.wowhead.com/npc=572/leprithus#comments

## Tests Performed:

- Watched each NPC spawn or despawn during events.  
- Followed Leprithus around and tested his formations.


## How to Test the Changes:


1. At 6:00 Server time go to Stormwind and see if the following civilians despawn:
Adam and Billy
Justin, Brandon, and Roman
Miss Danna and school children
Donna and William
Janey, Suzanne, and Lisan
Orphans that are running around at Orphanage
Karlee Chaddis, Paige Chaddis, and Gil
2. At 6:00 PM go to Eastvale Logging camp and see if they are despawned
3. At 8:00 PM check for Leprithus GUID 134020 and 134021.  They should both spawn.  The southern has waypoints
4. At 9:00 PM check for Ultham Ironhorn GUID 76265 has moved to in front of the corral
5. At 9:00 PM check for Ultham Ironhorn's ram have moved into the corral
6. At 9:00 PM check for Rotten Ghouls replacing fleshrippers and smugglers in graveyard in Moonbrook and next to Jangolode mine.
7. At 9:00 PM check Pyrewood village for them to convert to Moonrage Worgen. 
8. At 9:00 PM Check Moonrage Sentry's are hostile to alliance players
9.  At 6:00 AM all Stormwind civilians, Fleshrippers, smugglers, Pyrewood Village should return to normal.
10. At 6:00 AM any Leprithus that are still alive will despawn.

## Known Issues and TODO List:


- [The times for these events are based on websites in the source section and other obscure blogs.  If someone has more accurate info please let me know.]


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
